### ISSUE TITLE
"Backup File Disclosure" (10095): reports URL even if 404 

### ISSUE THREAD

**Issue by eoftedal** [zaproxy/zaproxy#4165]
After running an active scan on a domain, I saw active scan returning multiple findings for .bac, .backup, .bak, .jar, .log etc. even though the server returned a 404. The returned page was a standard jetty 404 page.

Expectation: 404 should exclude the result from findings and from the list of URLs in the context.

**Comment by eoftedal** [comment]
Response was:

```
HTTP/1.1 404 Not Found
Date: Mon, 18 Dec 2017 09:17:29 GMT
X-FRAME-OPTIONS: SAMEORIGIN
X-XSS-Protection: 1; mode=block
X-Content-Type-Options: nosniff
Pragma: no-cache
Content-Type: text/html;charset=ISO-8859-1
Cache-Control: must-revalidate,no-cache,no-store
Content-Length: 1393

<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=ISO-8859-1"/>
<title>Error 404 Not Found</title>
</head>
<body><h2>HTTP ERROR 404</h2>
<p>Problem accessing /example/file.old. Reason:
<pre>    Not Found</pre></p><hr /><i><small>Powered by Jetty://</small></i><br/>                                                
<br/>                                                
<br/>                                                
<br/>                                                
<br/>                                                
<br/>                                                
<br/>                                                
<br/>                                                
<br/>                                                
<br/>                                                
<br/>                                                
<br/>                                                
<br/>                                                
<br/>                                                
<br/>                                                
<br/>                                                
<br/>                                                
<br/>                                                
<br/>                                                
<br/>                                                

</body>
</html>
```

**Comment by kingthorin** [comment]
Is the 404 returned after a redirect?
Can you provide all of the alert details?

Could you enable debug logging https://github.com/zaproxy/zaproxy/wiki/FAQlogging and provide the log details for the "BackupFileDisclosure" scanner?

**Comment by thc202** [comment]
Was that using latest version of Active scanner rules (beta) add-on?

**Comment by eoftedal** [comment]
Yes, this was the latest version of that plugin

**Comment by eoftedal** [comment]
1. Run this server: https://gist.github.com/eoftedal/f074ae595da29e6224a7e417ebdad881
2. Start a blank session, and import this sequence: https://gist.github.com/eoftedal/a3a8b99764913bb792f5d59e7d266564
3. Run the sequence once from script console
4. Active scan the /email/ folder with the sequence enabled in active scan

It's not 100% reliable. If it doesn't work the frist time, start with a blank session and also restart the node server, and repeat steps 3 and 4.
Sometime it even reports SQL-injection even though there is no database...

**Comment by kingthorin** [comment]
It seems to be a DoS issue. If the server can't consistently return a response then of course clients (scanners or otherwise) might mishandle things.

If I run a full policy then the issue manifests easily. If I reduce to just the "Backup File Disclosure" plugin (at default/Medium strength) then no problem. If I increase to Insane strength then the issue manifests again (to a lesser extent).

**Comment by kingthorin** [comment]
Can someone else try this and confirm or deny what I experienced?

**Comment by thc202** [comment]
Sure.

**Comment by kingthorin** [comment]
@thc202 any luck?

**Comment by kingthorin** [comment]
Anyone?

**Comment by kingthorin** [comment]
Can this be closed? An explanation was provided 9months ago.

**Comment by thc202** [comment]
For the record, still plan to take a look at this, it just kept falling in the list...

**Comment by ovanekem** [comment]
I see this issue is still opened and I just encountered the same in a recent pentest using latest version. I also put the issues found as false positive as indeed if the web server returns an HTTP 404, it means it is not subject to any file backup disclosure.
Is there any follow up since last year on this ?

**Comment by black-snow** [comment]
I second this, here's some curl output:

```
> GET /xyz/index.bac HTTP/1.1
> Host: super.secret.tld
> User-Agent: curl/7.67.0
> Accept: */*
>
* Mark bundle as not supporting multiuse
< HTTP/1.1 404 404
< Date: Wed, 11 Dec 2019 08:53:08 GMT
< Server: Apache
< Content-Language: en
< Content-Length: 1099
< Content-Type: text/html;charset=utf-8
<
<!doctype html><html lang="en"><head><title>HTTP Status 404 – Not Found</title>...
```

**Comment by thc202** [comment]
Closing, addressed with the changes done for #6059.
